Source/WebCore/loader/ResourceMonitor.cpp

```diff
@@ -77,9 +77,11 @@ void ResourceMonitor::setDocumentURL(URL&& url)
 
     didReceiveResponse(m_frameURL, ContentExtensions::ResourceType::Document);
 
-    if (RefPtr iframe = dynamicDowncast<HTMLIFrameElement>(protectedFrame()->ownerElement())) {
-        if (auto& url = iframe->initiatorSourceURL(); !url.isEmpty())
-            didReceiveResponse(url, ContentExtensions::ResourceType::Script);
+    if (RefPtr frame = m_frame.get()) {
```

Comment:
Seems we can check `m_frame` before `didReceiveResponse` and return early if it is null